microsoft/testsuites/nested/common.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List, Optional
+
+from lisa import RemoteNode
+from lisa.schema import Node
+from lisa.tools import Qemu, Wget
+from lisa.tools.df import Df, PartitionInfo
+from lisa.util import SkippedException
+
+NESTED_VM_IMAGE_NAME = "image.qcow2"
+NESTED_VM_TEST_FILE_NAME = "message.txt"
+NESTED_VM_TEST_FILE_CONTENT = "Message from L1 vm!!"
+NESTED_VM_TEST_PUBLIC_FILE_URL = "http://www.github.com"
+NESTED_VM_REQUIRED_DISK_SIZE_IN_GB = 6
+
+
+def connect_nested_vm(
+    host: RemoteNode,
+    guest_username: str,
+    guest_password: str,
+    guest_port: int,
+    guest_image_url: str,
+    image_name: str = NESTED_VM_IMAGE_NAME,
+    image_size: int = NESTED_VM_REQUIRED_DISK_SIZE_IN_GB,
+    disks: Optional[List[str]] = None,
+) -> RemoteNode:
+    image_folder_path = get_partition_for_nested_image(host, image_size)
+
+    host.tools[Wget].get(
+        url=guest_image_url,
+        file_path=image_folder_path,
+        filename=image_name,
+        sudo=True,
+    )
+
+    # start nested vm
+    host.tools[Qemu].create_vm(guest_port, f"{image_folder_path}/{image_name}", disks)
+
+    # setup connection to nested vm
+    nested_vm = RemoteNode(Node(name="L2-vm"), 0, "L2-vm")
+    nested_vm.set_connection_info(
+        public_address=host.public_address,
+        username=guest_username,
+        password=guest_password,
+        public_port=guest_port,
+        port=guest_port,
+    )
+
+    return nested_vm
+
+
+def get_partition_for_nested_image(node: RemoteNode, size: int) -> str:
+    home_partition = node.tools[Df].get_partition_by_mountpoint("/home")
+    if home_partition and check_partition_capacity(home_partition, size):
+        return home_partition.mountpoint
+
+    mnt_partition = node.tools[Df].get_partition_by_mountpoint("/mnt")
+    if mnt_partition and check_partition_capacity(mnt_partition, size):
+        return mnt_partition.mountpoint
+
+    raise SkippedException(
```

Comment:
Why is it skipped by this condition? Is it possible there is any error?

Reply:
This is skipped if we are unable to find a partition with required capacity

Reply:
What's the reason not found? If the reason is the environment is broken by last test case, this case should fail to get attention. So, we need to think about what's situation here, if it's unknown, keep this test case fail, instead of skipping.

Reply:
The reason is that none of the disk has enough capacity. It was marked as skipped because this essentially means that the VM doesn't have the requirement to run the test.

Reply:
My concern is, if it's enough to differentiate not capable by design and by some unknown reason. Normally, if a test case returns skipped, most likely nobody looks the result again. So, we need to either make sure it's skipped by design or return exception for now. So we can check the reason later, if it returns exception.

Reply:
If you feel so, I will mark it as an Exception